extensions/wikia/AbPerfTesting/classes/Hooks.class.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Wikia\AbPerfTesting;
+
+class Hooks {
+
+	/**
+	 * Initialize performance experiments when MediaWiki starts the engine
+	 *
+	 * @param \Title $title
+	 * @param \Article $article
+	 * @param \OutputPage $output
+	 * @param \User $user
+	 * @param \WebRequest $request
+	 * @param \MediaWiki $wiki
+	 * @return bool it's a hook
+	 */
+	static function onAfterInitialize( \Title $title, \Article $article, \OutputPage $output, \User $user, \WebRequest $request, \MediaWiki $wiki ) {
+		global $wgABPerfTestingExperiments;
+		wfDebug( sprintf( "%s - checking experiments (with beacon ID set to '%s')...\n", __METHOD__, wfGetBeaconId() ) );
+
+		// loop through all registered experiments and run those matching criteria
+		foreach ( $wgABPerfTestingExperiments as $name => $experiment ) {
+			if ( Experiment::isEnabled( $experiment ) ) {
+				wfDebug( sprintf( "%s: starting '%s' experiment using %s class with %s params\n",
+					__METHOD__, $name, $experiment['handler'], json_encode( $experiment['params'] ) ) );
+
+				$reflector = new \ReflectionClass( $experiment['handler'] );
```

Comment:
Reflection used to be slow and not considered a best practice in production code, has anything changed in that matter? (I haven't tracked it thoroughly recently)